src/internal/operations/test/exchange_test.go

```diff
@@ -842,3 +845,333 @@ func testExchangeContinuousBackups(suite *ExchangeBackupIntgSuite, toggles contr
+
+		ctx, flush := tester.NewContext(t)
+		defer flush()
+
+		mb := evmock.NewBus()
+
+		restoreCfg.OnCollision = control.Copy
+
+		ro, _ := prepNewTestRestoreOp(
+			t,
+			ctx,
+			bod.st,
+			bo.Results.BackupID,
+			mb,
+			count.New(),
+			sel,
+			opts,
+			restoreCfg)
+
+		runAndCheckRestore(t, ctx, &ro, mb, -1, false)
+
+		// get all files in folder, use these as the base
+		// set of files to compare against.
+		contGC, err := acCont.GetContainerByName(ctx, userID, "", restoreCfg.Location)
+		require.NoError(t, err, clues.ToCore(err))
+
+		cIDs[path.ContactsCategory] = ptr.Val(contGC.GetId())
+
+		contM, err := acCont.GetItemsInContainerByCollisionKey(
+			ctx,
+			userID,
+			cIDs[path.ContactsCategory])
+		require.NoError(t, err, clues.ToCore(err))
+		maps.Copy(collKeys, contM)
+
+		// gc, err = acEvts.GetContainerByName(ctx, userID, "", restoreCfg.Location)
+		// require.NoError(t, err, clues.ToCore(err))
+
+		// restoredContainerID[path.EventsCategory] = ptr.Val(gc.GetId())
+
+		// m, err = acEvts.GetItemsInContainerByCollisionKey(
+		// 	ctx,
+		// 	userID,
+		// 	cIDs[path.EventsCategory])
+		// require.NoError(t, err, clues.ToCore(err))
+		// maps.Copy(collKeys, m)
+
+		mailGC, err := acMail.GetContainerByName(ctx, userID, api.MsgFolderRoot, restoreCfg.Location)
+		require.NoError(t, err, clues.ToCore(err))
+
+		mailGC, err = acMail.GetContainerByName(ctx, userID, ptr.Val(mailGC.GetId()), api.MailInbox)
+		require.NoError(t, err, clues.ToCore(err))
+
+		cIDs[path.EmailCategory] = ptr.Val(mailGC.GetId())
+
+		mailM, err := acMail.GetItemsInContainerByCollisionKey(
+			ctx,
+			userID,
+			cIDs[path.EmailCategory])
+		require.NoError(t, err, clues.ToCore(err))
+		maps.Copy(collKeys, mailM)
+	})
+
+	// skip restore
+
+	suite.Run("skip collisions", func() {
+		t := suite.T()
+
+		ctx, flush := tester.NewContext(t)
+		defer flush()
+
+		mb := evmock.NewBus()
+		ctr := count.New()
+
+		restoreCfg.OnCollision = control.Skip
+
+		ro, _ := prepNewTestRestoreOp(
+			t,
+			ctx,
+			bod.st,
+			bo.Results.BackupID,
+			mb,
+			ctr,
+			sel,
+			opts,
+			restoreCfg)
+
+		deets := runAndCheckRestore(t, ctx, &ro, mb, 0, false)
+
+		assert.Equal(
+			t,
+			int64(len(collKeys)),
+			ctr.Total(count.CollisionSkip),
+			"all attempted item restores should have been skipped")
+		assert.Zero(
+			t,
+			len(deets.Entries),
+			"no items should have been restored")
+
+		// get all files in folder, use these as the base
+		// set of files to compare against.
+		result := filterCollisionKeyResults(
+			t,
+			ctx,
+			userID,
+			cIDs[path.ContactsCategory],
+			giicbcker[string](acCont),
+			collKeys)
+
+		// m = checkCollisionKeyResults(t, ctx, userID, cIDs[path.EventsCategory], acEvts, collKeys)
+		// maps.Copy(result, m)
+
+		m := filterCollisionKeyResults(
+			t,
+			ctx,
+			userID,
+			cIDs[path.EmailCategory],
+			giicbcker[string](acMail),
+			collKeys)
+		maps.Copy(result, m)
+
+		assert.Len(t, result, 0, "no new items should get added")
```

Comment:
nit
```suggestion
		assert.Empty(t, result, "no new items should get added")
```

Reply:
also, this test is not sufficient to say we didn't duplicate or replace items since there's no item ID comparison happening

Reply:
I'll add support for this in a branch, and will link the pr when it's ready.